tests/test_sensor.py

```diff
@@ -262,3 +262,55 @@ def test_remove_callbacks(self, ss, sensors, add_callback, remove_callback):
         ss.remove(sensors[0])
         remove_callback.assert_not_called()
         add_callback.assert_not_called()
+
+
+class MyAgg(AggregateSensor):
+    def update_aggregate(self, updated_sensor, reading, old_reading):
+        return Reading(0, Sensor.Status.NOMINAL, 7)
+
+
+def test_aggregate_sensor(mocker, ss, sensors):
+    """Test operation of AggregateSensor.
+
+    From a simple derived class, mock out the _update_aggregate function, and
+    test that (1) the target is properly assigned, and (2) the update function
+    is called at all appropriate occasions.
+    """
```

Comment:
This is a bit of a test-everything test. I'd suggest splitting it up into separate tests within a TestAggregateSensor class - roughly one per comment you've got in the current test.